src/app/comments/create-comment/create-comment.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, Input } from '@angular/core';
+import { AngularFirestore } from '@angular/fire/firestore';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { Observable } from 'rxjs';
+import { AuthService } from 'src/app/core/auth.service';
+
+@Component({
+  selector: 'app-create-comment',
+  templateUrl: './create-comment.component.html',
+  styleUrls: ['./create-comment.component.scss']
+})
+export class CreateCommentComponent {
+  @Input() data: any;
+  user: Observable<any>;
+  comment: string;
+  uploading = false;
+
+  constructor( private afs: AngularFirestore, private snackBar: MatSnackBar, private auth: AuthService) {
+    this.auth.user$.subscribe(user => this.user = user);
+  }
+
+  submitComment = (): void => {
+    this.uploading = true;
+    this.afs.collection(`posts/${this.data.id}/comments`).add({
+      author: this.user,
+      comment: this.comment
+    })
+    /* Success message */
+    .then(() => {
+      this.snackBar.open('Upload successful!', 'Close', { duration: 3000 });
+      this.comment = ''; // This resets the comment form
+      this.uploading = false;
+    })
+    /* Error message */
+    .catch(() => {
```

Comment:
A general catch is fine enough, but maybe we could display the exception message / log it into the console... It would help a lot with integrating TravisCI and other testing tools.

Reply:
It works just fine for now though!

Reply:
This catch should display an error message using the snackbar. It's a popup at the bottom of the screen. We can always check the cause of the error in the network tab of the inspector.

Reply:
@CamilBouzidi looks to me like the catch notifies the user about the failure with a arrow function